### PR TITLE
feat(Core/Battleground): Added new config variables to battleground b…

### LINKED ISSUE #5098: Incorrect respawn timer for the restoration buff in Battlegrounds  [$15]
WHAT CLIENT DO YOU PLAY ON?

enUS

FACTION

Alliance
Horde

CONTENT PHASE:

Generic
1-19
20-29
30-39

CURRENT BEHAVIOUR:

Restauration buff respawn timer in battlegrounds is several minutes.
EXPECTED BLIZZLIKE BEHAVIOUR:

Restauration buff respawn timer in battlegrounds should be 20 seconds.

evidence link from wowhead:
https://classic.wowhead.com/guides/classic-wow-warsong-gulch-advanced-intro-rules-timers-wsg

tl;dr screenshot of the relevant part from the evidence link above:
https://i.gyazo.com/0b52b1922f39bec5afeefd3eade914f0.png
STEPS TO REPRODUCE THE PROBLEM:

    Join a battleground (for example, Warsong Gulch)
    Take a restauration buff (for exemple, inside the hut in Warsong Gulch)
    Notice how the respawn of the restauration buff takes several minutes instead of the expected 20 seconds.

EXTRA NOTES:

Spellid link for the restauration buff:

https://classic.wowhead.com/spell=23493/restoration
AC HASH/COMMIT:

chromiecraft/azerothcore-wotlk@9d4eebc
OPERATING SYSTEM:

Ubuntu 20.04
MODULES:

    mod-cfbg
    mod-duel-reset
    mod-desertion-warnings
    mod-ah-bot
    mod-eluna-lua-engine
    lua-LevelUpReward

OTHER CUSTOMIZATIONS:

None.
SERVER:

ChromieCraft

from

https://github.com/chromiecraft/chromiecraft/issues/325

<bountysource-plugin>

---
There is a **[$15 open bounty](https://www.bountysource.com/issues/97578960-incorrect-respawn-timer-for-the-restoration-buff-in-battlegrounds?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
…uffs. Corrected buff timers.

Fixed #5098.

## Changes Proposed:
-  Added newconfig variables to handle different buff timers.
-  Corrected buffs respawn timers.

## Issues Addressed:

- Closes  #5098.
- Closes https://github.com/chromiecraft/chromiecraft/issues/325

## Tests Performed:

- Not tested.

## How to Test the Changes:

```
Join a battleground (for example, Warsong Gulch)
Take a restauration buff (for exemple, inside the hut in Warsong Gulch)
Notice how the respawn of the restauration buff takes 20 seconds.
```

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Battlegrounds/Battleground.cpp [hunk 1]
```diff
@@ -1821,8 +1821,8 @@ void Battleground::HandleTriggerBuff(GameObject* gameObject)
 
     uint32 index = 0;
     for (; index < BgObjects.size() && BgObjects[index] != gameObject->GetGUID(); ++index);
-    if (BgObjects[index] != gameObject->GetGUID())
-        return;
+        if (BgObjects[index] != gameObject->GetGUID())
+            return;
```
Comment (FrancescoBorzi) [inline src/server/game/Battlegrounds/Battleground.cpp:1825]: the indentation is confusing here, can you please add brackets?

**Comment by FrancescoBorzi** [comment]
@UltraNix build failing :(

**Comment by Helias** [comment]
merge conflcits

**Comment by UltraNix** [comment]
Can be merged?

**Comment by FrancescoBorzi** [comment]
> Can be merged?

> Tests Performed:
> Not tested.

@UltraNix  sorry, we cannot merge changes that are not tested (unless it's very trivial ones). Can you please test it?

**Comment by UltraNix** [comment]
Not really.

**Comment by FrancescoBorzi** [comment]
> Not really.

then we need to wait for someone else to test this.

@azerothcore/testers can anyone please check this PR ?

**Comment by Hacaw** [comment]
@FrancescoBorzi I will test this, looks interesting.

**Comment by Hacaw** [comment]
TLDR; The fix works.

- Test1: 20 sec for healing buff Horde

https://user-images.githubusercontent.com/6963500/124762937-19464700-df3c-11eb-9293-4a145239e7c4.mp4

- Test2: 20 sec for healing buff Alliance

https://user-images.githubusercontent.com/6963500/124763079-409d1400-df3c-11eb-9681-52fdff445740.mp4

- Test3: adjusted the berserker buff at 20 seconds in the config file

https://user-images.githubusercontent.com/6963500/124771800-1c453580-df44-11eb-8d07-164aa3ee744f.mp4


I tested @UltraNix branch with a clean and fresh docker setup including database purged volumes and containers.

I only found an issue so far when the db-assembler script is ran I encountered this issue:
` ERROR 1062 (23000) at line 102496: Duplicate entry '1624655271618733500' for key 'version_db_world.PRIMARY'`
which points to my commit: 6557bd00f0dca5d2f612ac793911b7958029332a (could be only a local issue)

@FrancescoBorzi if there are any sql conflicts are the automated tests able to identify them ?
